src/common/noisers.py

```diff
@@ -153,6 +153,78 @@ def __call__(self, x):
     z = self._func(x)
     return z + self._random_state.normal(size=z.shape, scale=self._sigma)
 
+class DiscreteGaussianMechanism:
+  """Transforms a function using the discrete gaussian mechanism.
+
+  If f(x) = (Z[1], Z[2], ..., Z[k]), then returns a function that computes
+  (Z'[1], Z'[2], ..., Z'[k]), where
+      Z'[i] = Z[i] + Y[i],
+      Y[i] ~ N_Z(x | sigma),
+  and N_Z(x | sigma) is given by the probability mass function defined on the
+  integers that is given by
+      N_Z(x | sigma) ~ exp(-0.5 x^2 / sigma^2) / (sigma * sqrt(2 * pi))
```

Comment:
How about calling out x is any integer. And, shall we standardize N_Z to have sum=1?